meteor/client/ui/Settings/ShowStyle/VariantSettings.tsx

```diff
@@ -181,18 +246,101 @@ export const ShowStyleVariantsSettings = withTranslation()(
 			})
 		}
 
-		private renderShowStyleVariants() {
+		private confirmRemoveAllShowStyleVariants = (): void => {
 			const { t } = this.props
+			doModalDialog({
+				title: t('Remove all variants?'),
+				no: t('Cancel'),
+				yes: t('Remove'),
+				onAccept: () => {
+					this.removeAllShowStyleVariants()
+				},
+				message: (
+					<React.Fragment>
+						<p>{t('Are you sure you want to remove all variants in the table?')}</p>
+					</React.Fragment>
+				),
+			})
+		}
+
+		private removeAllShowStyleVariants = (): void => {
+			this.state.dndVariants.forEach((variant: ShowStyleVariant) => {
+				MeteorCall.showstyles.removeShowStyleVariant(variant._id).catch(logger.warn)
+			})
+		}
 
-			return this.props.showStyleVariants.map((showStyleVariant) => {
-				return (
-					<React.Fragment key={unprotectString(showStyleVariant._id)}>
+		private persistStateVariants = (): void => {
+			MeteorCall.showstyles
+				.reorderAllShowStyleVariants(this.props.showStyleBase._id, this.state.dndVariants)
+				.catch(logger.warn)
+		}
+
+		VariantItem = ({ index, showStyleVariant, moveVariantHandler }) => {
+			const ref = useRef<HTMLTableRowElement>(null)
+			const [{ handlerId }, drop] = useDrop<DragVariant, void, { handlerId: Identifier | null }>({
+				accept: ShowStyleDragDropTypes.VARIANT,
+				collect(monitor) {
+					return {
+						handlerId: monitor.getHandlerId(),
+					}
+				},
```

Comment:
This can be simplified to:
```suggestion
				collect: (monitor) => ({ handlerId: monitor.getHandlerId() }),
```